### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by blink1073**
All lines covered except in the generated `__extends` variable.

### CODE REVIEW

File: test/src/index.ts [hunk 1]
```diff
@@ -9,12 +9,341 @@
...
+} from 'phosphor-signaling';
+
+import {
+  MSG_LAYOUT_REQUEST, MSG_UPDATE_REQUEST, ResizeMessage, Widget, attachWidget
+} from 'phosphor-widget';
+
+import {
+  STACKED_PANEL_CLASS, StackedPanel
+} from '../../lib/index';
+
```
Comment (sccolbert) [inline test/src/index.ts]: 2 blank lines after last import

File: test/src/index.ts [hunk 2]
```diff
@@ -9,12 +9,341 @@
...
+        expect(StackedPanel.currentWidgetProperty.get(panel)).to.eql(widget0);
+        StackedPanel.currentWidgetProperty.set(panel, widget1);
+        expect(panel.currentWidget).to.eql(widget1);
+      });
+
+    });
+
+    describe('#onChildAdded()', () => {
+
+      it('should hide the new node', () => {
```
Comment (sccolbert) [inline test/src/index.ts]: should hide the new child*

File: test/src/index.ts [hunk 3]
```diff
@@ -9,12 +9,341 @@
...
+        panel.children = [widget0, widget1];
+        panel.moveChild(1, 0);
+        expect(panel.messages.indexOf('child-moved')).to.not.be(-1);
+      });
+
+    });
+
+    describe('#onAfterShow()', () => {
+
+      it('should be invoked just after the panel is made visible', () => {
```
Comment (sccolbert) [inline test/src/index.ts]: This should test that it posts a `'layout-request'` message.

File: test/src/index.ts [hunk 4]
```diff
@@ -9,12 +9,341 @@
...
+        var panel = new LogPanel();
+        attachWidget(panel, document.body);
+        panel.hidden = true;
+        panel.hidden = false;
+        expect(panel.messages.indexOf('after-show')).to.not.be(-1);
+      });
+
+    });
+
+    describe('#onAfterAttach()', () => {
```
Comment (sccolbert) [inline test/src/index.ts]: This should test that it posts a `'layout-request'` message.

File: test/src/index.ts [hunk 5]
```diff
@@ -9,12 +9,341 @@
...
+
+      it('should be invoked just after the panel is attached', () => {
+        var panel = new LogPanel();
+        attachWidget(panel, document.body);
+        expect(panel.messages.indexOf('after-attach')).to.not.be(-1);
+      });
```
Comment (sccolbert) [inline test/src/index.ts]: one blank line after closing `it`.

File: test/src/index.ts [hunk 6]
```diff
@@ -9,12 +9,341 @@
...
+        var msg = new ResizeMessage(50, 55);
+        sendMessage(panel, msg);
+        expect(widget.offsetRect.width).to.not.be(50);
+      });
+
+    });
+
+    describe('#onUpdateRequest()', () => {
+
+      it('should be invoked when an update is requested', () => {
```
Comment (sccolbert) [inline test/src/index.ts]: This should test that the `currentWidget` geometry actually gets updated.

Comment (sccolbert) [inline test/src/index.ts]: It should also be a no-op if invisible.

File: test/src/index.ts [hunk 7]
```diff
@@ -9,12 +9,341 @@
...
+        panel.hidden = true;
+        sendMessage(panel, MSG_LAYOUT_REQUEST);
+        expect(widget.offsetRect.width).to.be(0);
+        panel.currentWidget = widget;
+        sendMessage(panel, MSG_LAYOUT_REQUEST);
+        expect(widget.offsetRect.width).to.not.be(110);
+      });
 
-    it('should always pass', () => {
+      it('should be propagated to a parent', () => {
```
Comment (sccolbert) [inline test/src/index.ts]: should send a `'layout-request'` to the parent

**Comment by blink1073** [comment]
Comments addressed, not sure why Travis can't install node all of a sudden...

File: test/src/index.ts [hunk 8]
```diff
@@ -9,12 +9,371 @@
...
+        panel.children = [widget0, widget1];
+        panel.moveChild(1, 0);
+        expect(panel.messages.indexOf('child-moved')).to.not.be(-1);
+      });
+
+    });
+
+    describe('#onAfterShow()', () => {
+
+      it('should be invoked just after the panel is made visible', (done) => {
```
Comment (sccolbert) [inline test/src/index.ts]: should send an `'update-request'` message.

File: test/src/index.ts [hunk 9]
```diff
@@ -9,12 +9,371 @@
...
+        var panel = new LogPanel();
+        attachWidget(panel, document.body);
+        panel.hidden = true;
+        panel.hidden = false;
+        expect(panel.messages.indexOf('after-show')).to.not.be(-1);
+        requestAnimationFrame(() => {
+          expect(panel.messages.indexOf('layout-request')).to.not.be(-1);
```
Comment (sccolbert) [inline test/src/index.ts]: test for `'update-request'`, and it should be sent immediately, not posted.

File: test/src/index.ts [hunk 10]
```diff
@@ -9,12 +9,371 @@
...
+          done();
+        });
+      });
+
+    });
+
+    describe('#onAfterAttach()', () => {
+
+      it('should be invoked just after the panel is attached', (done) => {
```
Comment (sccolbert) [inline test/src/index.ts]: "should post a `'layout-request'` message"

**Comment by blink1073** [comment]
You merged before I could address the last few comments, want a follow-up?

**Comment by sccolbert** [comment]
Nah, I took care of it.

**Pull request merged**